### PR TITLE
vice: update to 3.7.1.

### PR DESCRIPTION
**Pull request by ypsvlq**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: srcpkgs/vice/files/desktop-install.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh -e
+dir=$DESTDIR/usr/share/applications/
+[ "$1" = install ] && mkdir -p $dir && shift && for f; do cp $f $dir; done
```
Comment (paper42) [inline srcpkgs/vice/files/desktop-install.sh:3]: Is there a reason why you didn't just add xdg-utils to hostmakedepends?

Comment (ypsvlq) [reply srcpkgs/vice/files/desktop-install.sh:3]: @paper42 it didn't install properly into DESTDIR, I'm not sure if there's some XDG_ env var that needs to be set for it to work

**Comment by ypsvlq** [comment]
bump

File: srcpkgs/vice/files/desktop-install.sh (hunk 1 above)
Comment (paper42) [reply srcpkgs/vice/files/desktop-install.sh:3]: Did you try XDG_DATA_DIRS?

File: srcpkgs/vice/template [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
-	$(vopt_enable ffmpeg  external-ffmpeg)
-	--disable-option-checking
+	$(vopt_enable ffmpeg external-ffmpeg)
 	--enable-cpuhistory
 	--enable-x64
 	--disable-pdf-docs
-	AR=${AR}"
-hostmakedepends="bdftopcf flex mkfontdir perl pkg-config xa
- glib-devel autoconf texinfo automake dos2unix"
+	--libdir=/usr/lib$XBPS_TARGET_WORDSIZE
```
[the remark this answers is not in the document]
Comment (ypsvlq) [reply srcpkgs/vice/template:16]: Not specifying it there causes 
```
=== configuring in src/resid (/builddir/vice-3.6.1/src/resid)
configure: running /bin/sh ./configure '--prefix=/usr' --disable-option-checking  '--sysconfdir=/etc' '--sbindir=/usr/bin' '--bindir=/usr/bin' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--localstatedir=/var' '--host=x86_64-unknown-linux-gnu' '--build=x86_64-unknown-linux-gnu' '--libdir=NONE/lib64' '--disable-sdlui2' '--without-sdlsound' '--enable-native-gtk3ui' '--enable-desktop-files' '--with-pulse' '--enable-external-ffmpeg' '--enable-cpuhistory' '--enable-x64' '--disable-pdf-docs' 'build_alias=x86_64-unknown-linux-gnu' 'host_alias=x86_64-unknown-linux-gnu' 'CC=cc' 'CFLAGS=-fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe    -fdebug-prefix-map=/builddir/vice-3.6.1=.' 'LDFLAGS=-Wl,-z,relro -Wl,-z,now -Wl,--as-needed    ' 'CPPFLAGS=
 ' 'CPP=cpp' 'CXX=g++' 'CXXFLAGS=-fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe    -fdebug-prefix-map=/builddir/vice-3.6.1=.' 'PKG_CONFIG=pkg-config' '--disable-option-checking' '--srcdir=.' --cache-file=
configure: error: expected an absolute directory name for --libdir: NONE/lib64
configure: error: ./configure failed for src/resid
=> ERROR: vice-3.6.1_1: do_configure: '${configure_script} ${configure_args}' exited with 1
```

File: srcpkgs/vice/files/desktop-install.sh (hunk 1 above)
Comment (ypsvlq) [reply srcpkgs/vice/files/desktop-install.sh]: Ah, I did but it needs directories to exist already

**Comment by ypsvlq** [comment]
bump

**Comment by ypsvlq** [comment]
bump i guess

File: srcpkgs/vice/template [hunk 3]
```diff
@@ -1,26 +1,24 @@
 # Template file for 'vice'
 pkgname=vice
-version=3.5
+version=3.6.1
 revision=1
 build_style=gnu-configure
 configure_args="
```
Comment (zlice) [inline srcpkgs/vice/template:6]: do a `configure_args+=` and see if that works w/o the `--libdir`?

File: srcpkgs/vice/template [hunk 4]
```diff
@@ -32,26 +30,24 @@ makedepends="
...
+}
 
-pre_configure() {
-	# __u_char is not portable...
-	sed -i 's,__u_char,unsigned char,g' src/*.[ch]
-	sed -i 's/lib64/lib/g' configure.ac
-	autoreconf -fi
+pre_install() {
+	# required for xdg-desktop-menu to work
+	mkdir -p $DESTDIR/usr/share/{applications,desktop-directories}
```
Comment (classabbyamp) [inline srcpkgs/vice/template:52]: use vmkdir

File: srcpkgs/vice/template [hunk 5]
```diff
@@ -32,26 +30,24 @@ makedepends="
...
 desc_option_ffmpeg="Support video recording via FFmpeg"
 
-CFLAGS="-fcommon"
+export XDG_DATA_DIRS=$DESTDIR/usr/share
 
-case "$XBPS_TARGET_MACHINE" in
-	*-musl) configure_args+=" --disable-nls";;
-esac
+post_patch() {
+	sed -i -e 's/\$(XDG_DESKTOP_MENU) install/$(XDG_DESKTOP_MENU) install --mode system/' src/arch/gtk3/data/unix/Makefile.in
```
Comment (classabbyamp) [inline srcpkgs/vice/template:47]: use vsed

**Pull request merged**